src/utils/ucc_coll_utils.c

```diff
@@ -291,3 +291,62 @@ void ucc_coll_str(const ucc_coll_task_t *task, char *str, size_t len)
         strncat(str, tmp, left);
     }
 }
+
+ucc_ep_map_t ucc_ep_map_create_reverse(ucc_rank_t size)
+{
+    ucc_ep_map_t map = {.type           = UCC_EP_MAP_STRIDED,
+                        .ep_num         = size,
+                        .strided.start  = size - 1,
+                        .strided.stride = -1};
+    return map;
+}
+
+static inline int ucc_ep_map_is_reverse(ucc_ep_map_t *map)
+{
+    return (map->type == UCC_EP_MAP_STRIDED) &&
+           (map->strided.start == map->ep_num - 1) &&
+           (map->strided.stride == -1);
+}
+
+ucc_status_t ucc_ep_map_create_inverse(ucc_ep_map_t map, ucc_ep_map_t *inv_map)
+{
+    ucc_ep_map_t inv;
+    ucc_rank_t   i, r;
+    ucc_rank_t   max_rank;
+
+    if (ucc_ep_map_is_reverse(&map)) {
+        inv = map;
```

Comment:
the function intention is to inverse the given map and here  your seting the same map to inv.  isn't possible to call inverse only if the map is not reverse?

Reply:
THe intent was to keep the client (tl/ucp) code clean. In case of bidirectional alg, tl/ucp ring always allocates just 2 rings 1 default + 1 reverse. I need to have inverse map for all rings to implement the alg. In this case when my ring are straight forward i could just  omit call to inverse in tl/ucp since it evaluates to the original ring. However, when we had custom rings used in the same alg (example was HCM, where rings would be topo dependent) then inverse compute is necessary. But the code stays the same, we can just pass a new "map" to the alg.